packages/components/src/navigation/index.js

```diff
@@ -121,6 +121,96 @@ const Navigation = ( { activeItemId, children, data, rootTitle } ) => {
 	);
 };
 
+export function Navigation2( { children } ) {
+	const [ activeLevel, setActiveLevel ] = useState( 'root' );
+	const [ slideOrigin, setSlideOrigin ] = useState( 'left' );
```

Comment:
This points to a real challenge with this approach because the top level component won't know about its children. The initial `slideOrigin` won't necessarily be "left" because we may navigate from a link on the page that will represent an animation with `slideOrigin` "right".

Previously, we calculated a `isNavigatingBack` variable to solve this issue. Maybe `Navigation` analyses its children to determine hierarchy position?

Reply:
Very good point!
I wanted to do something like `isNavigatingBack` but didn't have time to explore it.
Though I definitely didn't consider that the navigation could load "in media res", which requires a more involved handling of the slide origin default.

Reply:
Actually, thinking about it, the initial `slideOrigin` could be omitted altogether, since it's not technically used (there is no animation on mount).

But regardless, with your implementation of `NavigationLevel` (I really shouldn't have used "Navigation2" as a name, now anything looks like "Navigation to level" 😄) the back button takes care of it automatically.

Reply:
>Previously, we calculated a isNavigatingBack variable to solve this issue. Maybe Navigation analyses its children to determine hierarchy position?

I like this idea.  Though it would mean the children would have to be in the "correct" order to determine this.